src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -51,6 +51,36 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
         return builder.AddParameter(name, () => value, publishValueAsDefault, secret);
     }
 
+    /// <summary>
+    /// Creates a new <see cref="ParameterResource"/> that has a generated value using the <paramref name="value"/>.
+    /// </summary>
+    /// <remarks>
+    /// The value will be saved to the app host project's user secrets store when <see cref="DistributedApplicationExecutionContext.IsRunMode"/> is <c>true</c>
+    /// and the lifetime of the resource is <see cref="ContainerLifetime.Persistent"/>.
+    /// </remarks>
+    /// <param name="builder">Distributed application builder</param>
+    /// <param name="name">Name of the parameter</param>
+    /// <param name="value"></param>
```

Comment:
The `value` parameter is a bit odd since it is only used on the first run. After the value is persisted once, calling `AddPersistentParameter` again with a different `value` is ignored, since it reads it from user secrets.

Maybe calling this `initialValue` or similar may help.